crates/router/src/db/merchant_account.rs

```diff
@@ -244,28 +216,23 @@ impl MerchantAccountInterface for Store {
         &self,
         merchant_id: &str,
     ) -> CustomResult<bool, errors::StorageError> {
-        let delete_func = || async {
-            let conn = connection::pg_connection_write(self).await?;
-            storage::MerchantAccount::delete_by_merchant_id(&conn, merchant_id)
-                .await
-                .map_err(Into::into)
-                .into_report()
-        };
+        let conn = connection::pg_connection_write(self).await?;
 
-        #[cfg(not(feature = "accounts_cache"))]
-        {
-            delete_func().await
-        }
+        let _merchant_account = storage::MerchantAccount::find_by_merchant_id(&conn, merchant_id)
+            .await
+            .map_err(Into::into)
+            .into_report()?;
```

Comment:
this can be moved under the feature flag right? This db call needs to be performed only during the `accounts_cache` feature flag